web-api/src/persistence/postgres/cases/getFullEligibleCasesForTrialSession.ts

```diff
@@ -0,0 +1,95 @@
+import { getDbReader } from '@web-api/database';
+import {
+  CASE_STATUS_TYPES,
+  SESSION_TYPES,
+  TrialSessionTypes,
+} from '@shared/business/entities/EntityConstants';
+import { purgeDynamoKeys } from '@web-api/persistence/dynamo/helpers/purgeDynamoKeys';
+import { transformNullToUndefined } from '@web-api/persistence/postgres/utils/transformNullToUndefined';
+import { rawCaseEntity } from '@web-api/persistence/postgres/cases/mapper';
+import { ServerApplicationContext } from '@web-api/applicationContext';
+import { getPrivatePractitionersOnCase } from '@web-api/persistence/dynamo/practitioners/getPrivatePractitionersOnCase';
+import { getIrsPractitionersOnCase } from '@web-api/persistence/dynamo/practitioners/getIrsPractitionersOnCase';
+import { IrsPractitioner } from '@shared/business/entities/IrsPractitioner';
+import { PrivatePractitioner } from '@shared/business/entities/PrivatePractitioner';
+
+export const getFullEligibleCasesForTrialSession = async ({
+  applicationContext,
+  limit,
+  trialCity,
+  sessionType,
+}: {
+  applicationContext: ServerApplicationContext;
+  limit: number;
+  trialCity: string;
+  sessionType: TrialSessionTypes;
+}) => {
+  const dbCases = await getDbReader(async reader => {
+    let query = reader
+      .selectFrom('dwCase')
+      .selectAll()
+      .where('preferredTrialCity', '=', trialCity)
+      .where('status', '=', CASE_STATUS_TYPES.generalDocketReadyForTrial)
+      .where(eb =>
+        eb.and([
+          eb('automaticBlocked', 'is not', true),
+          eb('blocked', 'is not', true),
+          eb.not(
+            eb.exists(sq =>
+              sq
+                .selectFrom('dwCase as c2')
+                .select('c2.leadDocketNumber')
+                .where('c2.preferredTrialCity', '=', trialCity)
+                .whereRef('c2.leadDocketNumber', '=', 'dwCase.leadDocketNumber')
+                .where(qb =>
+                  qb.or([
+                    qb('c2.automaticBlocked', '=', true),
+                    qb('c2.blocked', '=', true),
+                  ]),
+                ),
+            ),
+          ),
+        ]),
+      )
+      .limit(limit);
+
+    if (sessionType !== SESSION_TYPES.hybrid) {
+      query = query.where('procedureType', '=', sessionType);
+    }
+    return await query.execute();
```

Comment:
Are the columns included in the query they only thing that differs between this and the other persistence function for getting eligible cases? To my previous comment, our rules for case eligibility are now baked into these queries -- is there some way we can share those rules between the two queries so that if those rules change, we don't have to remember to change them in both places?

E.g. maybe we there's some way to dynamically choose between the two different column sets (all vs. subset)?

Curious your thoughts

Reply:
There is, it was just breaking type safety. Likely worth spending more time finding a workaround